### ISSUE TITLE
NewBlockHashesMessage.getBlockIdentifiers should parse if not already parsed.

### ISSUE THREAD

**Issue by mvanotti** [ethereum/ethereumj#441]
The code for [getBlockIdentifiers](https://github.com/ethereum/ethereumj/blob/9cf08dafe474014c577c0636df6cfa48022828bc/ethereumj-core/src/main/java/org/ethereum/net/eth/message/NewBlockHashesMessage.java#L65-L67) should have the following logic:

`if (!parsed) parse();`

To avoid possible NPE.

**Comment by Nashatyrev** [comment]
Yep, it worked just because of logging level which invoked `toString()`. Thanks for catching this!
